src/pal/inc/pal.h

```diff
@@ -5971,129 +5941,82 @@ class PAL_NoHolder
-        PAL_TryExcept(__Body::Run, __Body::Handler, &__handlerData, &__fExecuteHandler);	\
-    PAL_NoHolder __exceptionHolder;                                     \
-    if (__exception && __fExecuteHandler)                               \
-    {                                                                   \
-        PAL_CatchHolder __catchHolder(__exception);                     \
-        __exception = NULL;
-
-#define PAL_EXCEPT_FILTER(filter) PAL_EXCEPT(filter(ExceptionPointers, pvParam))
+            throw;                                                              \
+        }
 
-// Executes the handler if the specified exception code matches
-// the one in the exception. Otherwise, returns EXCEPTION_CONTINUE_SEARCH.
-#define PAL_EXCEPT_IF_EXCEPTION_CODE(dwExceptionCode) \
-        PAL_EXCEPT(((ExceptionPointers->ExceptionRecord->ExceptionCode == dwExceptionCode)?EXCEPTION_EXECUTE_HANDLER:EXCEPTION_CONTINUE_SEARCH))
 
-#define PAL_FINALLY                                                     \
-        }                                                               \
-        static EXCEPTION_DISPOSITION Filter(                            \
-            EXCEPTION_POINTERS *ExceptionPointers,                      \
-            PAL_DISPATCHER_CONTEXT *DispatcherContext,                  \
-            void *pvHandlerData)                                        \
-        {                                                               \
-            DEBUG_OK_TO_RETURN_BEGIN(PAL_FINALLY)                       \
-            return EXCEPTION_CONTINUE_SEARCH;                           \
-            DEBUG_OK_TO_RETURN_END(PAL_FINALLY)                         \
-        }                                                               \
-    };                                                                  \
-    BOOL __fExecuteHandler;                                             \
-    _Unwind_Exception *__exception =                                    \
-        PAL_TryExcept(__Body::Run, __Body::Filter, &__handlerData, &__fExecuteHandler); \
-    PAL_ExceptionHolder __exceptionHolder(__exception);                 \
-    {
-
-#define PAL_ENDTRY                              \
-    }                                           \
-    if (__exception)                            \
-    {                                           \
-        __exceptionHolder.SuppressRelease();    \
-        PAL_Leave(PAL_BoundaryBottom);          \
-        _Unwind_Resume(__exception);            \
-    }                                           \
+#define PAL_EXCEPT_FILTER(filter) PAL_EXCEPT(filter(&ex.ExceptionPointers, __param))
+
+#define PAL_FINALLY \
+    }; \
+    const bool isFinally = true; \
+    auto finallyBlock = [&]() \
+    {                       
+
+#define PAL_ENDTRY \
+    }; \
+    if (isFinally) \
+    { \
+        try \
+        { \
+            tryBlock(__param);        \
```

Comment:
Oops, seems like VS has reformatted that piece of code under my hands. I had all the backslashes aligned to the same column.